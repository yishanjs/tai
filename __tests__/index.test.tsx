/**
 * Copyright (c) 2019 Yishan Authors
 *
 * All right reserved
 */

import * as React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

import App from '@/pages/index';

describe('Index', () => {
  it('should shows"Yishan \'s typescript&next starter"', async () => {
    const app = shallow(<App />);

    expect(app.find('h1').text()).toEqual("Yishan typescript&next starter");
  });

  it('should match snapshot', async () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  });
});
