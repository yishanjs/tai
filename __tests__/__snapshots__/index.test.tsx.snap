// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should match snapshot 1`] = `
<div>
  <div
    className="bp3-navbar"
  >
    <div
      className="bp3-navbar-group bp3-align-left"
    >
      <div
        className="bp3-navbar-heading"
      >
        Blueprint
      </div>
      <div
        className="bp3-navbar-divider"
      />
      <button
        className="bp3-button bp3-minimal"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-home"
          icon="home"
        >
          <svg
            data-icon="home"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              home
            </desc>
            <path
              d="M2 10v5c0 .55.45 1 1 1h3v-5h4v5h3c.55 0 1-.45 1-1v-5L8 4l-6 6zm13.71-2.71L14 5.59V2c0-.55-.45-1-1-1s-1 .45-1 1v1.59L8.71.29C8.53.11 8.28 0 8 0s-.53.11-.71.29l-7 7a1.003 1.003 0 001.42 1.42L8 2.41l6.29 6.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Home
        </span>
      </button>
      <button
        className="bp3-button bp3-minimal"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-document"
          icon="document"
        >
          <svg
            data-icon="document"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              document
            </desc>
            <path
              d="M9 0H3c-.55 0-1 .45-1 1v14c0 .55.45 1 1 1h10c.55 0 1-.45 1-1V5L9 0zm3 14H4V2h4v4h4v8z"
              fillRule="evenodd"
            />
          </svg>
        </span>
        <span
          className="bp3-button-text"
        >
          Files
        </span>
      </button>
    </div>
  </div>
  <h1>
    Yishan typescript&next starter
  </h1>
</div>
`;
